packages/smart-vaults/src/vault/SmartVault.sol

```diff
@@ -403,19 +399,20 @@ contract SmartVault is IAccount, Ownable, UUPSUpgradeable, MultiSignerAuth, ERC1
     )
         internal
         view
-        returns (uint256 validationData)
+        returns (uint256)
     {
-        if (!MerkleProof.verify(signature.merkleProof, signature.merkleTreeRoot, userOpHash_)) {
-            revert InvalidMerkleProof();
-        }
+        bool isValidMerkleProof = MerkleProof.verify(signature.merkleProof, signature.merkleTreeRoot, userOpHash_);
 
         if (signature.signatures.length > 1) {
             if (!MerkleProof.verify(signature.lightMerkleProof, signature.lightMerkleTreeRoot, lightHash_)) {
-                revert InvalidMerkleProof();
+                isValidMerkleProof = false;
```

Comment:
can replace the inner if with:
```
isValidMerkleProof &&= MerkleProof.verify(signature.lightMerkleProof, signature.lightMerkleTreeRoot, lightHash_);
```

Reply:
would be slightly more costly for the happy case?

Reply:
are you sure? `if` requires a jump which is(?) more expensive than just `and`ing the answers together

Reply:
ah forgot about the if jump